starter-code/src/views/NewBeer.jsx

```diff
@@ -0,0 +1,101 @@
+import React, { Component } from 'react';
+import { Redirect } from 'react-router-dom'
+import axios from 'axios';
+
+class NewBeer extends Component {
+
+  state = {
+    newBeer : {},
+    beerAdded : false,
+    error : false
+  }
+
+  onChange(e) {
+    let fieldVal =  e.target.name === 'attenuation_level' ? parseInt(e.target.value) : e.target.value;
+
+    let newBeer = {...this.state.newBeer}
+    newBeer[e.target.name] = fieldVal;
+
+    this.setState({ newBeer })
+  }
+
+  onSubmit(e) {
+    e.preventDefault();
+    this.sendDataToAPI(this.state.newBeer);
+  }
+
+  sendDataToAPI(payload) {
+    axios.post("https://ih-beers-api2.herokuapp.com/beers/new", payload)
+    .then((results) => {
+      this.setState({ beerAdded : true })
+    }).catch((err) => {
+      console.log("Error : ", err)
+      this.setState({ error : true })
+    })
+  }
+
+  render() {
+
+    if (!!this.state.error) {
+      return (
+        <div className="page newbeer">
+          <p>Oops... something went wrong !</p>
+        </div>
+      )
+    }
+
+    if (!!this.state.beerAdded) {
+      return (<Redirect to='/' />)
+    }
+
+    return (
+      <div className="page newbeer">
+        <form method="post" onChange={(e) => this.onChange(e)} onSubmit={(e) => this.onSubmit(e)}>
```

Comment:
Here you are creating an anonymous function and you are not passing any extra value apart from the event.
This pattern is not recommended.

https://reactjs.org/docs/handling-events.html

> The problem with this syntax is that a different callback is created each time the LoggingButton renders. In most cases, >this is fine. However, if this callback is passed as a prop to lower components, those components might do an extra re->rendering. We generally recommend binding in the constructor or using the class fields syntax, to avoid this sort of >performance problem.